docs/installation/advanced_topics.asciidoc

```diff
@@ -107,6 +134,203 @@ View of the opened security events:
 
+==== SQL
+
+When `type=sql` PacketFence uses MySQL script/batch mode to manually build the SQL query including the execution of multiple statements. This provides complete query control as well as the ability to manage the SQL session and the SQL transaction. This is the preferred mode where SQL optimization is needed to execute complex queries, or for those more comfortable with raw (non-abstract) SQL.
+
+  sql=SELECT * FROM sponsors;
+
+Multiline block (heredoc) is required when executing multiple statements. Each statement should be terminated with a semi-color ";".
+
+NOTE: SQL execution exits on the first error and returns the result set of the last successful statement.
+
+The following attributes are available when using `type=sql`: [options="compact"]
+
+* `bindings`: A comma delimited list of ordered bindings to send to the SQL script (ex: `bindings=tenant_id,start_date,end_date,cursor,limit`). See Bindings below.
+* `cursor_type`: _[node|field_multi_field]_ Adds a cursor binding to the sql script that implements pagination of the results. The cursor is automatically handled in the administration interface, but its use in the `sql` requires special attention. If omitted the default `none` is used. More information about cursors is provided below. There are 2 types of cursors:
+ ** `cursor_type=field`: Use a single field (column or alias) for the cursor.
+ ** `cursor_type=multi_field`: Use multiple fields (columns or aliases) for the cursor.
+ ** `cursor_type=offset`: Use integer based offset for the cursor.
+ ** `cursor_type=none`: No cursor is used.
+* `cursor_default`: The default cursor used to conditionally query the results for the first page. On subsequent pages this is replaced with the results from N+1 row of the previous page, meaning the cursor for page 2 (with `default_limit=25`) will contain the value from the column of the 26th row from the previous page.
+* `cursor_field`: A comma delimited list of fields (columns) used for pagination.
+* `default_limit`: The default limit passed into the bindings of the SQL script. When `has_limit=enabled` the reports' user can override the default with a manual selection.
+* `sql`: Either a single MySQL query, or a multi line block of statements within a heredoc (see Heredoc below).
+
+==== Bindings
+
+The `bindings` attribute defines an ordered comma delimited list of columns (or aliases) that are made available to the `sql` script. There is no limit with the number of bindings that can used and a binding can be repeated more than once.
+
+The available bindings are: [options="compact"]
```

Comment:
`[options="compact"]` is an asciidoc list setting; move it on its own line, right before the list definition.